### ISSUE TITLE
paho-mqtt not found?

### ISSUE THREAD

**Issue by MennoLabs** [FilipDem/Domoticz-iROBOT-ROOMBA-plugin#8]
pi@domoticz:~ $ sudo service roomba status
● roomba.service - Roomba mqtt client
   Loaded: loaded (/etc/systemd/system/roomba.service; enabled; vendor preset: enabled)
   Active: active (running) since Sun 2024-06-23 22:39:37 CEST; 1min 45s ago
 Main PID: 481 (python3)
    Tasks: 2 (limit: 2176)
   CGroup: /system.slice/roomba.service
           └─481 /usr/bin/python3 /home/pi/domoticz/plugins/Roomba/roomba --configfile /home/pi/domoticz/plugins/Roomba/conf

Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library
Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library
Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library
Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library
Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library
Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library
Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library
Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library
Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library
Jun 23 22:41:22 domoticz python3[481]: Please install paho-mqtt 'pip install paho-mqtt' to use this library

pi@domoticz:~ $ sudo pip3 install paho-mqtt
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Requirement already satisfied: paho-mqtt in /usr/local/lib/python3.7/dist-packages (2.1.0)

**Comment by FilipDem** [comment]
Perhaps a conflict sudo versus not sudo?
I assume also that all is using python3 (pip3 vs pip).

**Comment by MennoLabs** [comment]
Nope:

pi@domoticz:~ $ service roomba start
==== AUTHENTICATING FOR org.freedesktop.systemd1.manage-units ===
Authentication is required to start 'roomba.service'.
Authenticating as: ,,, (pi)
Password: 


In the mean time, these  "Please install paho-mqtt 'pip install paho-mqtt' to use this library" lines filled my daemon.log to the point where the complete disk was full.

Maybe some symlink is in order to point the service to the proper location?

**Comment by MennoLabs** [comment]
My paho-mqtt is in /usr/local/lib/python3.7/dist-packages/paho (Raspbian Buster), does that offer a clue?
Python version 3.7.3.

**Comment by FilipDem** [comment]
Hi, 
I am also using Python 3.7.3 running on Buster.
paho is also installed in the same folder as you mentioned.

Did you already tried using the command prompt? Open a python3 prompt and use "paho.mqtt.client".
I also assume you installed Domoticz "natively"... not using a Docker container, right?

F

**Comment by MennoLabs** [comment]
pi@domoticz:~ $ python3
Python 3.7.3 (default, Mar 23 2024, 16:12:05) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> paho.mqtt.client
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'paho' is not defined
>>> 

Native, no Dockers.

**Comment by FilipDem** [comment]
This means that it is not installed correctly. Nothing to do with Domotica or the plugin

**Comment by MennoLabs** [comment]
I never meant to blame either the plugin or Domoticz, but I do wonder why it's not working as intended when installing with `sudo pip3 install paho-mqtt`, as per instructions?

**Comment by FilipDem** [comment]
I know. No worry. Perhaps uninstall and reinstall...

Op di 9 jul. 2024 22:32 schreef MennoLabs ***@***.***>:

**Comment by MennoLabs** [comment]
Just tried that (while you were typing your reply ;-)), no success.

**Comment by FilipDem** [comment]
I would advise to "google" on sth like "paho.mqtt.client not found"... I
see there are several people with a similar problem as you...

Op di 9 jul 2024 om 22:36 schreef MennoLabs ***@***.***>:

**Comment by MennoLabs** [comment]
Yes, tried that, but the lucky tip has not yet materialised from that search.
Will try some more tomorrow.

**Comment by jjnh** [comment]
I used the plugin successfully on my Domoticz 2023.1 until I upgraded it to 2025.2

My setup:
- Domoticz 2025.2 in a docker using the official container 
- Domoticz is a fresh container install with retention of the previous user config and a restore of the Domoticz DB
- The docker image has python 3.11 installed by default
- Roomba setup and connected
- Test script for password etc. works absolutely fine
- When I try to run the process manually I receive the error that paho-mqtt is not installed

What I tried
- Upgraded the plugin files with the most recent from github here 
- Uninstalled and reinstalled Python and the paho-mqtt components
- Verified robot vacuum is there and pingable
- test script for password --> works
- Python reinstall --> worked
- Python checked --> works
- installed dependencies and checked that paho-mqtt is there --> works
- Sudo is not part of the Docker image and also was never needed to start the process previously

When starting the plugin now it still shows paho-mqtt missing and then starts dumping lines "Please install paho-mqtt 'pip install paho-mqtt' to use this library".
Still I cannot get the iRobot connected anymore.

Any ideas what might be going on here and how I can fix this?

**Comment by FilipDem** [comment]
This is not an issue of the plugin... I recently also moved to Domoticz in a docker container... 
Indeed no sudo, just installing mqtt-paho... Stupid remark: you install it inside the docker container, right?
Don't forget to add it in the customstart.sh otherwise at every restart of the container, it is gone...

**Comment by jjnh** [comment]
No stupid remark. Good to check. 

All actions above were done inside the Docker container indeed. 
Everything worked until after the upgrade. I put alle dependencies in the customstart.sh when I did the original install.
Mqtt-paho is installed. I verified that. 

It's just that after starting the script throws the constant error that it's not.
The service also will not start.

**Comment by FilipDem** [comment]
Sorry I am wrong... I don't install the paho-mqtt in the domoticz container.
The roomba service runs completely outside the container. I created a dedicate dservice outside the domoticz container, but in a python environment. 
be aware that also an mqtt broker needs to be available (mosquitto).
F

**Comment by jjnh** [comment]
So the Roomba service can run independently from where Domoticz and/or Mosquitto are running. I wasn't sure that was possible.

I'll give it a whirl and let you know. Thanks!

**Comment by jjnh** [comment]
That worked. I now installed the MQTT service in the local OS an have the plugin running in the Domoticz container.
All good again.

Thanks for the support!!

**Comment by FilipDem** [comment]
Thanks for the contribution! 
Just for your info: this is basic MQTT: you have a broker and the Roomba sends information to the broker. The plugin connects to the broker also to fetch information (or send information). So there is no need that the broker or the roomba service is inside the domoticz container... There is in fact even no reason to do so...
I assume I can close the topic?

**Comment by jjnh** [comment]
You are welcome 😊 

Yes, you can close the topic.
